### PR TITLE
Fix for schema imports with new tests validating

### PR DESCRIPTION
**Pull request by Deeg-Kim**
*Description of changes:*

* Changed the parsing of id within import to be a string instead of symbol
* Added new test cases to verify that type imports are happening
 * Added new test cases to verify that imported types would be used during validation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/software/amazon/ionschema/IonSchemaImportTest.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                .loadSchema("schema/Customer.isl")
+                .getTypes()
+                .asSequence()
+                .associateBy({ it.name() })
+        assertTrue(types.contains("Customer"))
+        assertTrue(types.contains("positive_int"))
+    }
+
+    @Test
+    fun validateType() {
```
Comment (pbcornell) [inline test/software/amazon/ionschema/IonSchemaImportTest.kt]: This appears to be redundant with the tests defined in data/test/schema/Customer.isl.  The preference here is for the file-based tests (so they can be leveraged to verify the behavior of other Ion Schema impls).  Are you seeing additional value in having this test method?

File: test/software/amazon/ionschema/IonSchemaImportTest.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        val type = iss.loadSchema("schema/Customer.isl").getType("Customer")!!
+        assertTrue(type.isValid(ION.singleValue("{ firstName: \"Phil\", lastName: \"Collins\" }")))
+        assertFalse(type.isValid(ION.singleValue("{ firstName: \"Phil\", middleName: \"Billy\" }")))
+    }
+
+    @Test
+    fun validateAgainstImportedType() {
```
Comment (pbcornell) [inline test/software/amazon/ionschema/IonSchemaImportTest.kt]: same comment as above for this test method

File: data/test/schema/Customer.isl [hunk 3]
```diff
@@ -8,6 +11,7 @@ type::{
     firstName: { type: string, occurs: required },
     middleName: string,
     lastName: { type: string, codepoint_length: range::[min, 7], occurs: required },
+    age: { type: positive_int, occurs: optional },
```
Comment (pbcornell) [inline data/test/schema/Customer.isl]: `occurs: optional` is the default for a field.  Please change this line to `age: positive_int,`

File: test/software/amazon/ionschema/IonSchemaImportTest.kt (hunk 2 above)
Comment (Deeg-Kim) [reply test/software/amazon/ionschema/IonSchemaImportTest.kt]: Although we are effectively just testing the validation logic, I believe there is value in testing the validation against an imported type  -- this verifies that we do import, we can validate against it. More generally, this is a way to test that the type has been imported successfully.

I do agree that the test above may be less valuable.

Comment (pbcornell) [reply test/software/amazon/ionschema/IonSchemaImportTest.kt]: Agreed, there's value in testing validation against an imported type.  However, the changes you've proposed in `data/test/schema/Customer.isl` do exactly that.  A successful test run against that file requires loading the Customer type, which in turn imports `positive_int` from `CommonTypes.isl`, and then verifying that the age field validates as expected.  It exercises the exact same set of API calls and codepaths that this test does.

Do we need unit tests that exercise API behavior?  Absolutely.  But we should avoid creating unit tests that duplicate those within `data/test`.

**Pull request merged**